fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAutoCompleteResultCursor.java

```diff
@@ -169,12 +170,325 @@ private void performLookup() throws IOException {
             return;
         }
         long startTime = System.nanoTime();
-        lookupResults = suggester.getCount() > 0
-                        ? suggester.lookup(query, Collections.emptySet(), limit, true, highlight)
-                        : Collections.emptyList();
+
+        lookupResults = lookup().skip(skip);
         if (timer != null) {
             timer.recordSinceNanoTime(LuceneEvents.Events.LUCENE_AUTO_COMPLETE_SUGGESTIONS_SCAN, startTime);
-            timer.increment(LuceneEvents.Counts.LUCENE_SCAN_MATCHED_AUTO_COMPLETE_SUGGESTIONS, lookupResults.size());
+        }
+    }
+
+    @SuppressWarnings("squid:S3776") // Cognitive complexity is too high. Candidate for later refactoring
+    @Nullable
+    private String searchAllMaybeHighlight(String text, Set<String> matchedTokens, @Nullable String prefixToken, boolean highlight) {
```

Comment:
I think this method is worth unit tests.

Reply:
Sure. I added `LuceneAutoCompleteResultCursorTest` and put some unit tests of this functionality in here.

We may want to refactor the tokenize-and-search logic, at some point, because I could see us wanting to re-use this code somewhere else, but I didn't do that here.